src/lightning_lite/utilities/data.py

```diff
@@ -0,0 +1,424 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import inspect
+import os
+from collections import OrderedDict
+from contextlib import contextmanager
+from functools import partial
+from typing import Any, Callable, Dict, Generator, Iterable, Optional, Set, Tuple, Type, Union
+
+from torch.utils.data import BatchSampler, DataLoader, IterableDataset, Sampler
+
+from lightning_lite.utilities.enums import LightningEnum
+from lightning_lite.utilities.exceptions import MisconfigurationException
+from lightning_lite.utilities.rank_zero import rank_zero_warn
+from lightning_lite.utilities.seed import pl_worker_init_function
+
+
+class _WrapAttrTag(LightningEnum):
+    SET = "set"
+    DEL = "del"
+
+    def __call__(self, *args):
+        if self == self.SET:
+            fn = setattr
+        else:
+            fn = delattr
+        return fn(*args)
+
+
+def has_iterable_dataset(dataloader: DataLoader) -> bool:
+    return hasattr(dataloader, "dataset") and isinstance(dataloader.dataset, IterableDataset)
+
+
+def has_len(dataloader: Union[DataLoader, Iterable]) -> bool:
+    """Checks if a given Dataloader has ``__len__`` method implemented i.e. if it is a finite dataloader or
+    infinite dataloader."""
+    try:
+        # try getting the length
+        if len(dataloader) == 0:
+            rank_zero_warn(
+                f"`{dataloader.__class__.__name__}` returned 0 length. Please make sure this was your intention."
+            )
+        has_len = True
+    except (TypeError, NotImplementedError):
+        has_len = False
+
+    if has_len and has_iterable_dataset(dataloader):
+        rank_zero_warn(
+            "Your `IterableDataset` has `__len__` defined."
+            " In combination with multi-process data loading (when num_workers > 1),"
+            " `__len__` could be inaccurate if each worker is not configured independently"
+            " to avoid having duplicate data."
+        )
+    return has_len
+
+
+def _update_dataloader(dataloader: DataLoader, sampler: Union[Sampler, Iterable]) -> DataLoader:
```

Comment:
Removed argument "mode" from this function, as lite knows no "running stage". No other changes.